File: Implementierung/Backend/src/main/java/edu/kit/informatik/model/userdata/interactions/Category.java
package edu.kit.informatik.model.userdata.interactions;

import edu.kit.informatik.model.DataObject;
import edu.kit.informatik.model.User;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.sql.Timestamp;

/**
 * Entität einer Kategorie
 *
 * @author uekai
 * @author ugqbo
 * @version 1.0
 */
@Getter
@NoArgsConstructor
@Entity
@Table(name = "categories")
public class Category extends DataObject {

    private String name;

    /**
     * Konstruktor zum Erstellen eines Objektes
     * @param user {@link User}
     * @param name Name
     * @param createdAt {@link Timestamp} der Erstellung
     * @param updatedAt {@link Timestamp} der letzten Änderung
     */
    public Category(User user, String name, Timestamp createdAt, Timestamp updatedAt) {
        super(user, createdAt, updatedAt);
        this.name = name;
    }

    /**
     * Setzen des Namens
     * @param name name
     */
    public void setName(String name) {
        this.name = name;
    }
}

File: Implementierung/frontend/paschi/src/model/userdata/interactions/Category.ts
import {User} from "@/model/User";
import {Quality} from "@/model/userdata/interactions/Quality";
import {DataObject} from "@/model/DataObject";

export class Category extends DataObject {

  private readonly _user: User;
  private _name: string;

  constructor(id: string | undefined, localId: number, user: User, name: string) {
    super(id, localId);
    this._user = user;
    this._name = name;
  }

  hasQuality(): boolean {
    return false;
  }

  getQuality(): Quality | undefined {
    return undefined;
  }

  get user(): User {
    return this._user;
  }

  get name(): string {
    return this._name;
  }

  set name(value: string) {
    this._name = value;
    this.update();
  }
}
